lib/src/DateRangePicker/DateRangePickerToolbar.tsx

```diff
@@ -33,51 +35,54 @@ interface DateRangePickerToolbarProps
   setCurrentlySelectingRangeEnd: (newSelectingEnd: 'start' | 'end') => void;
 }
 
-export const DateRangePickerToolbar: React.FC<DateRangePickerToolbarProps> = ({
-  date: [start, end],
-  toolbarFormat,
-  isMobileKeyboardViewOpen,
-  toggleMobileKeyboardView,
-  currentlySelectingRangeEnd,
-  setCurrentlySelectingRangeEnd,
-  startText,
-  endText,
-  toolbarTitle = 'SELECT DATE RANGE',
-}) => {
-  const utils = useUtils();
-  const classes = useStyles();
+export const DateRangePickerToolbar: React.FC<DateRangePickerToolbarProps> = withDefaultProps(
```

Comment:
What's the advantage of makeStyles+withDefaultProps over withStyles? I think that the codebase should be as close as possible to the mono repository. It will be less migration work to do when merging the git repositories. It's also less room for mistake, less confusion for new contributors etc.

Reply:
It is easier to migrate from the existed codebase and much easier to solve typing issues from injected `classes` prop.

I am not tent to rewrite everything to look the same as in the core repository. For example:  sure today everybody using `makeStyles` instead of `withStyles` because it is better for all cases. Just do the same

Reply:
Ok so the issue is with the TypeScript definitions. Can the typing we use in the mono-repository be used to solve the problem?

Reply:
No really, I am sure that `default props` should not be a part of the theme. Because it is not a theme and we have already started discussion of moving them to separate provider, like `Configuration`. 

So decoupling styles and theme already should help for potential refactorings. And overall I am not seeing any advantages in `withStyles` over `withDefaultProps`, it is just more friendly to an existing codebase, faster and easier for props type inference

Reply:
A couple of thoughts:

- 1. We have an opportunity to empower a simpler migration once we change the component structure strategy, later on, by using a single approach now.
- 2. Assuming we have a ConfigProvider, we could consider moving the default prop logic to this new object, however, it would hurt an issue with theming. it's relatively frequent to change default props that are only style-related: `disableRipple`, `color="secondary"`, `headlineMapping`, etc.
- 3. If we decouple, we should design the new API to minimize mistakes & inconsistencies. Right now, it's helped by bundling all the features under a single high-order component.
- 4. In https://github.com/mui-org/material-ui/pull/15023, we explore a potential migration path to the a hook API. I think that the conclusion was that it didn't worth the effort. 
- 5. Looking at how the code look like now and how it would look like in v5, I see a high level of chance that it will go in the opposite direction: default props would move from a hoc to a hook, styles will move from a hook to a hoc.

Reply:
> So decoupling styles and theme already should help for potential refactorings. And overall I am not seeing any advantages in withStyles over withDefaultProps, it is just more friendly to an existing codebase, faster and easier for props type inference

Going back to your questions
- prop type inference is definitely a concern. Higher-order components are hard to type. However, we had to solve this problem in the past for the developers. I don't know this part very well, I believe/hope we did a great job in here, without any significant downside :)
- the advantage of withStyles over withDefaultProps is that it will enable the current API documentation generation stack to output the correct information, without the need for custom work. It will also enable the `getClasses` testing tool to take the component, without custom work.
- regarding the performance, I wouldn't expect any difference. I think that the perf win in mui-org/material-ui#15023 was related to the removal of a React component, nothing else, which we are already paying with `withDefaultProps` that introduce a new component.